src/qt/bitcoin.cpp

```diff
@@ -309,9 +311,8 @@ void BitcoinApplication::startThread()
 
 void BitcoinApplication::parameterSetup()
 {
-    // Default printtoconsole to false for the GUI. GUI programs should not
-    // print to the console unnecessarily.
-    gArgs.SoftSetBoolArg("-printtoconsole", false);
+    gArgs.SoftSetBoolArg("-printtoconsole", SERVER_ARGS_OPTIONS.printtoconsole_default);
```

Comment:
what is the point of overwriting the arg here, but keeping the default value as `!args.GetBoolArg("-daemon", false)` where it is parsed? It would be cleaner to not do any overwriting here and simply use printtoconsole_default as default value.

Reply:
Besides https://github.com/bitcoin/bitcoin/blob/2ce271c84d7ac5cee0cb79f5f04662b8b7c42f5b/src/init.cpp#L894 we also have https://github.com/bitcoin/bitcoin/blob/2ce271c84d7ac5cee0cb79f5f04662b8b7c42f5b/src/bitcoin-wallet.cpp#L67 with a different default value.

Reply:
bitcoin-wallet is a completely separate binary that isn't even modified in this PR. I fail to see how this is relevant to the discussion?

Reply:
@MarcoFalke 
> It would be cleaner to not do any overwriting here and simply use printtoconsole_default as default value.

Yes, you right. But at the parsing place the `SERVER_ARGS_OPTIONS` is out of the scope. It is possible to bring it into the scope by adding a data member to the `ArgsManager` but it will make a diff bigger. Is it worth it?

Reply:
Why add a data member to ArgsManager? Just pass it into `InitLogging` by reference?

Reply:
> Just pass it into `InitLogging` by reference?

What about `interfaces::Node::initLogging()`?

Reply:
> what is the point of overwriting the arg here, but keeping the default value as `!args.GetBoolArg("-daemon", false)` where it is parsed? It would be cleaner to not do any overwriting here and simply use printtoconsole_default as default value.

It will change the current behavior, as for `bitcoind` the `printtoconsole_default` must be disabled when the `-daemon` option is set.

Reply:
@MarcoFalke 

Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/19471#issuecomment-822027233).